### PR TITLE
Update tests for changes in `@types/unist`

### PR DESCRIPTION
**Pull request by GuiltyDolphin**
### Initial checklist

*   [X] I read the support docs 
*   [X] I read the contributing guide 
*   [X] I agree to follow the code of conduct 
*   [X] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [X] If applicable, I’ve added docs and tests

### Description of changes

Updates to use the latest version of `@types/unist`.

cc @wooorm @ChristianMurphy

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "index.js"
   ],
   "dependencies": {
-    "@types/unist": "^2.0.0",
+    "@types/unist": "^2.0.5",
```
Comment (GuiltyDolphin) [inline package.json:50]: @ChristianMurphy Should I keep this at `2.0.0` as you suggest on unified?

File: test.js [hunk 2]
```diff
@@ -183,7 +183,7 @@ test('unist-util-visit', function (t) {
     st.end()
 
     /**
-     * @param {Node} node
+     * @param {Node & { value: string}} node
```
Comment (ChristianMurphy) [inline test.js:186]: this should probably be a Literal https://github.com/DefinitelyTyped/DefinitelyTyped/blob/18043fe2428e3e6bfe5c10d46852befe1996b6dd/types/unist/index.d.ts#L95-L97
```suggestion
     * @param {Literal} node
```

File: test.js [hunk 3]
```diff
@@ -435,7 +435,14 @@ test('unist-util-visit', function (t) {
 
   t.test('should visit added nodes', function (st) {
     var tree = remark().parse('Some _emphasis_, **importance**, and `code`.')
-    var other = remark().use(gfm).parse('Another ~~sentence~~.').children[0]
+
+    // Unified doesn't (yet) let us specify parse result type, so all
+    // we know is that it's a node, but we know it is a parent, so we
```
Comment (ChristianMurphy) [inline test.js:440]: Right, to specific a result type, unified would need to know the contents of the markdown ahead of time.
And would potentially need a second implementation of the parser inside the typings to solve what node type would be returned.

```suggestion
    // Unified doesn't know parse result type,
    // all we know is that it's a node, but we know it is a parent, so we
```

File: package.json (hunk 1 above)
Comment (ChristianMurphy) [reply package.json:50]: yes, the minimum version in range please

```suggestion
    "@types/unist": "^2.0.0",
```

File: test.js (hunk 2 above)
Comment (GuiltyDolphin) [reply test.js:186]: Ah yeah - I was going to use `Literal` but the typechecker in my head said that `unknown != string`, oops. Thanks.

**Review by ChristianMurphy — APPROVED** [review]
Thanks @GuiltyDolphin!

**Review by wooorm — APPROVED** [review]

**Pull request merged**